tests/gabbits/create-snippet.yaml

```diff
@@ -0,0 +1,116 @@
+common:
+  - &datetime_regex /\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d+\+\d{2}:\d{2}/
+  - &slug_regex /[a-zA-Z0-9]+/
+
+fixtures:
+  - XSnippetApi
+
+tests:
+  - name: create a new snippet
+    POST: /v1/snippets
+    request_headers:
+      content-type: application/json
+    data:
+      title: Hello, World!
+      syntax: python
+      content: print('Hello, World!')
+      tags:
+        - spam
+        - eggs
+    status: 200
+    response_headers:
+      content-type: application/json
+    response_json_paths:
+      $.id: *slug_regex
+      $.title: Hello, World!
+      $.syntax: python
+      $.created_at: *datetime_regex
+      $.updated_at: *datetime_regex
+      $.tags.`sorted`:
+        - eggs
+        - spam
+      $.changesets.`len`: 1
+      $.changesets[0].version: 0
+      $.changesets[0].content: print('Hello, World!')
+      $.changesets[0].created_at: *datetime_regex
+      $.changesets[0].updated_at: *datetime_regex
+
+  - name: create a new snippet (no tags)
+    POST: /v1/snippets
+    request_headers:
+      content-type: application/json
+    data:
+      title: Hello, World!
+      syntax: python
+      content: print('Hello, World!')
+    status: 200
+    response_headers:
+      content-type: application/json
+    response_json_paths:
+      $.id: *slug_regex
+      $.title: Hello, World!
+      $.syntax: python
+      $.created_at: *datetime_regex
+      $.updated_at: *datetime_regex
+      $.tags: []
+      $.changesets.`len`: 1
+      $.changesets[0].version: 0
+      $.changesets[0].content: print('Hello, World!')
+      $.changesets[0].created_at: *datetime_regex
+      $.changesets[0].updated_at: *datetime_regex
+
+  - name: create a new snippet (no title)
+    POST: /v1/snippets
+    request_headers:
+      content-type: application/json
+    data:
+      content: print('Hello, World!')
+      syntax: python
+      tags:
+        - spam
+        - eggs
+    status: 200
+    response_headers:
+      content-type: application/json
+    response_json_paths:
+      $.id: *slug_regex
+      $.title:
+      $.syntax: python
+      $.created_at: *datetime_regex
+      $.updated_at: *datetime_regex
+      $.tags.`sorted`:
+        - eggs
+        - spam
+      $.changesets.`len`: 1
+      $.changesets[0].version: 0
+      $.changesets[0].content: print('Hello, World!')
+      $.changesets[0].created_at: *datetime_regex
+      $.changesets[0].updated_at: *datetime_regex
+
```

Comment:
I have another idea for the test:

1. Test that snippet can be created with minimum required information (i.e. just `content` as input).